tests/test_docs.py

```diff
@@ -0,0 +1,34 @@
+import doctest
+import pytest
+
+from pathlib import Path
+
+
+@pytest.fixture(autouse=True)
+def base_dir():
+    return Path(__file__).parent.parent
+
+
+def test_readme(base_dir):
+    readme_path = base_dir / "README.md"
+    assert (
+        doctest.testfile(str(readme_path), module_relative=False).failed == 0
+    )
+
+
+def test_modules(base_dir):
+    for module in base_dir.glob("python/**/*.py"):
+        assert doctest.testfile(str(module), module_relative=False).failed == 0
+
+
+def test_rust_sources(base_dir):
+    for f in base_dir.glob("src/**/*.rs"):
```

Comment:
Can you give an example of a Python doctest in the Rust source code? Or what you might expect that to look like?

Reply:
You're right, there actually aren't any.